### DISCUSSION TITLE
🎉 Charts for all - expanding access to project insights

### DISCUSSION THREAD

**Discussion by 2percentsilk** [community/community#19658] [Enterprise]
We're excited to announce that all projects (beta) users can access custom current charts to help visualize their work! Head over to the Insights tab for your projects to try it out 👀 

If you're on a Free plan, you'll have access to up to 2 saved charts for your private projects and unlimited configurations for public projects!

We're also expanding access to time-based charts to allow organizations to visualize trends over time 📈   
- All Enterprise organizations have been enabled and will have unlimited access to current and time-based charts.
- Team organizations are being onboarded over the next couple weeks and time-based charts will be available as a *preview* feature.
  -  Good news - If you're already using projects (beta) for your Team, time-based charts should be available as a preview to you *right now*!

Thank you for all of your feedback during the limited beta and we hope you'll continue to share your thoughts with us

**Comment by JhonnyLusonode** [comment]
Oh nice! can´t wait!

I belive it would be a great idea putting some more achivements for example bug hunter when you report a bug on github or something related to that. 
That would make the community more active trying to find some bugs/vulnerabilities on github which leads to a more secure environment.

What are your thoughts on this? @2percentsilk

**Reply by 2percentsilk** [reply]
Hi there - I'd encourage you to share this feedback in the [Profile discussions topic](https://github.com/github-community/community/discussions/categories/profile) so that the Achievements team can see it!

**Comment by omaskery** [comment]
Hi, really excited about this feature set, thanks for this work 🎉! Unfortunately, we don't have seem to have access to time-based charts, even though [we're an organization](https://github.com/infinite-worlds) on the Team plan that has an existing Project?

**Reply by jayspadie** [reply]
Hi @omaskery, thanks for getting in touch to let us know.  I see that infinite-worlds is on a Team plan, and our records show you do have an existing project.  I'll follow up on our side to see if there were any issues with this particular org or other priorities applied for early Team org onboarding to the Insights platform (i.e., with identifying active projects (beta) users).  Sorry for the wait, but these should be coming soon regardless.  Please reply to this thread in a couple of weeks if time-based charts continue to be missing.  Thanks again.

**Reply by jayspadie** [reply]
Hey @omaskery , would you mind checking again and let us know?  We've been investigating and I flipped a flag that should expose historical time-based Insights charts in your org now.  We may have missed a few orgs in our onboarding automation, so thank you for bringing this to our attention!

**Reply by omaskery** [reply]
Hi @jayspadie I have the default CFD chart available now, hurrah! Thanks for looking into this for us :) Have a good day!

**Reply by jayspadie** [reply]
Great!  Thanks for confirming @omaskery.  Now we just need to identify and fix any other orgs that we may have missed. 😄

**Comment by democat3457** [comment]
Is it possible to have the x-axis as a title and the y-axis as a custom field value? Also, what about sorting the x-axis by descending or ascending y-axis? Our use case is that we have a leaderboard-style format for some cards in our project, and we would really like to show a visual comparison/ranking between those cards.

**Reply by den4iksiniak** [reply]
- [x] ``

**Reply by 2percentsilk** [reply]
Unfortunately, we don't support any of those customizations currently but will keep them in mind as we continue to improve upon the space!

**Comment by samctrumpet** [comment]
Hey @2percentsilk, love this feature and I can see it being really useful!

Is it possible currently to include archived cards on the charts? E.g we move our cards to Done and then archive them at the end of the sprint, but they don't appear in the chart.

**Reply by 2percentsilk** [reply]
Not as of yet but your feedback is heard! Out of curiosity, would you want ALL archived charts to appear or do you have some notion of "Archived within X days" that you'd like to configure?

**Reply by samctrumpet** [reply]
Hey @2percentsilk thanks for your reply!

I'm guessing when a card is archived, the date it was archived also gets stored in the process. So if we looked back over a specific date range we would see any cards archived in that range too.

**Comment by danielprehealth** [comment]
Love this feature! Any target timeline for the time-based charts to become more widely available, including to organizations not yet on a paid Team plan? Would love to have access to this!

**Reply by 2percentsilk** [reply]
At this time, there is no timeline to expand time-based charts to free plans. If you would like to utilize time-based charts, you will need to upgrade your organization to a Team or Enterprise Cloud plan.

**Reply by danielprehealth** [reply]
Hi @2percentsilk, we just upgraded our [organization](https://github.com/Pre-Health) to the paid Team plan to gain access to the time-based charts. It's been a couple days but we still don't have access to the Time-Based Charts in the Insights Tab under the Insights tab. Could you help?

**Reply by danielprehealth** [reply]
@jayspadie are you able to help unlock the time-based charts for our [organization](https://github.com/Pre-Health)?

**Reply by 2percentsilk** [reply]
Sorry for the delay @danielprehealth ! I'm going to file a ticket with our team to look into what's happening here. Out of curiosity, can you create a new project and see if that new one perhaps works? And to be extra clear, can you see current charts and just can't create time based charts in the Insights tab for your project?

**Reply by danielprehealth** [reply]
Thank you for the help! I just tried creating a new project, and time is still absent from the x-axis selections.

![image](https://user-images.githubusercontent.com/83095894/186022169-4ba44d05-0a85-4ed0-9862-089a45bdb5e6.png)

**Reply by danielprehealth** [reply]
And yes, I can create charts for my existing project just fine inside the Insights tab, but just don't have the option to use time-based charts. It's not an option under "Feature Preview" either.

**Reply by 2percentsilk** [reply]
OK thanks for verifying all that. Can you go to https://github.com/organizations/pre-health/settings/features and verify if Insights is turned on?

The page should look like this
![Screen Shot 2022-08-23 at 10 05 37 AM](https://user-images.githubusercontent.com/3174849/186220382-2c760b0c-b3cf-4b95-adf1-ec7bad2dc721.png)

**Reply by danielprehealth** [reply]
Yes, it was already turned on
![image](https://user-images.githubusercontent.com/83095894/186221854-62e63c6c-7962-4c22-9060-bbdf299f405c.png)

**Reply by jayspadie** [reply]
Hi @danielprehealth, thanks for your patience here.  The team is still investigating why the feature wasn't enabled as it should have been, but it's been manually enabled now.  Please let us know if you can see historical time-based charts now.  Thanks.

**Reply by danielprehealth** [reply]
It works now! This is awesome, thank you, Jay! Couple points of feedback:
1. Would be nice to be able to select which options are charted in "Group By." For example, in the below chart, it'd be great if I could hide "Backlog" and "Archive" when all I'm trying to do is visualize the current sprint status.
![image](https://user-images.githubusercontent.com/83095894/186265979-96d8f4ed-f457-4268-abe1-910229f6f3fc.png)

2. Sum of a Field when doing historical charts is broken.
![image](https://user-images.githubusercontent.com/83095894/186266118-7e73ca3e-3a4b-4153-a3bb-0722300eb743.png)

**Reply by jayspadie** [reply]
Great.  Thanks for confirming.  With regard to the other feedback:

1.  You'll probably find that you can filter what you want using the filter bar, then save that as part of your chart configuration.  With your example, you could specify `iteration:@current` or some other specific iteration.  Alternatively, you could exclude status with `-status:Backlog,Archive`.  Keep in mind though that filtering, even for historical charts, filters on the current item state.  So, for example, if you choose to exclude items with Backlog and Archive status, then it will chart all others over time that are currently match that filter, regardless of their status in the past.
2. Yup, you found a bug that I was just chatting about with the Insights team here.  Apparently custom column names containing some special characters, like brackets in your example, aren't handled well.  If you want to use the sum aggregate and willing to try renaming your column, you could probably work around it until it's fixed.  Sorry about that.

**Reply by danielprehealth** [reply]
1. That does we want, thanks! And noted about how filtering works - that makes sense.
2. I changed from bracket to parentheses, and that fixed it, thanks!

Awesome work, loving these features, thanks!

**Reply by jayspadie** [reply]
Perfect, thanks!  Also we have a fix coming out soon (likely later today) that will correctly handle Y-axis aggregate column names containing brackets.  The same code was used for group_by column names with brackets, but it was missed for aggregate column names.  As an aside, there can still be problems with column names containing an emoji, but not with column values containing an emoji.  Thanks again.

**Reply by alexsteixeira** [reply]
> Hi @danielprehealth, thanks for your patience here. The team is still investigating why the feature wasn't enabled as it should have been, but it's been manually enabled now. Please let us know if you can see historical time-based charts now. Thanks.

Same problem here. We just upgraded our [organization](https://github.com/2na8) to the Team plan (paid) but we still do not have access to the time based charts.

**Reply by Jaytxo** [reply]
We have the same problem. We already upgraded to a paid "Team" plan, but we can't use the historical charts with the x-asis time option.

**Comment by solyasa** [comment]
good idea.
but trading view provide chart library. We can use Trading view website for chart
[Technical Analysis Charts](https://www.tradingview.com/HTML5-stock-forex-bitcoin-charting-library/?feature=technical-analysis-charts)

**Comment by jdnichollsc** [comment]
Hey folks, this is awesome!!
Is there any chart to see the deadline and efforts of the team?

**Comment by slinna** [comment]
For those looking for more advanced analytics for GitHub projects, I just wanted to let you know that [Screenful](https://screenful.com/github) now supports all GitHub Projects (classic and new) in its analytics. With Screenful, you can track things like: 

- Trends (e.g. story points completed over time)
- Custom fields (both x-axis and y-axis, compare multiple fields)
- 16 chart types including, burndowns, multiline charts, multi-bar charts, stacked bar charts etc. 
- Forecasts (what the likely release date based on historical velocity)

You can learn more from the [Tour page](https://screenful.com/tour/insights) or start a free trial via the [GitHub Marketplace](https://github.com/marketplace/analytics-reports).

**Comment by beingminimal** [comment]
@2percentsilk when github pro users will get access for time based historical chart support in github project insights?

**Comment by sergiinvestor** [comment]
These expanded charts are a great step toward better project insights, especially the addition of time-based visualizations. It's really useful to see trends like commit activity and workflow evolution over time.

From my experience, combining these charts with external tools can sometimes provide an even more comprehensive perspective. For example, tools like [Gitlights](https://github.com/marketplace/gitlights-analytics) offer additional breakdowns, such as developer rankings and commit distributions, which help in understanding both individual and team contributions in detail.


<img width="1239" alt="Captura de pantalla 2024-12-17 a las 17 17 32" src="https://github.com/user-attachments/assets/392a063a-2eff-48d4-b9f1-896ec6539751" />
